contractsV5/KyberNetwork.sol

```diff
@@ -675,10 +675,86 @@ contract KyberNetwork is Withdrawable, Utils, IKyberNetwork, ReentrancyGuard {
         }
     }
     
+    event HandleFees(address indexed recipient, uint fees);
+
     function handleFees(TradeData memory tradeData) internal returns(bool) {
         // create array of reserves receiving fees + fee percent per reserve
         // fees should add up to 100%.
         // send total fee amount to fee handler with reserve data.
+
+        address[] memory eligibleReserves = new address[](tradeData.numFeePayingReserves);
+        uint[] memory rebatePercentages = new uint[](tradeData.numFeePayingReserves);
+
+        // Updates reserve eligibility and rebate percentages
+        updateEligibilityAndRebates(eligibleReserves, rebatePercentages, tradeData);
+
+        // Sending platform fee to taker platform
+        (bool success, ) = tradeData.input.platformFeeWei != 0 ?
+            tradeData.input.platformWallet.call.value(tradeData.platformFeeWei)("") :
+            (true, bytes(""));
+        require(success, "Transfer platform fee to taker platform failed");
+        emit HandleFees(tradeData.input.platformWallet, tradeData.platformFeeWei);
+
+        // Send total fee amount to fee handler with reserve data.
+        require(
+            feeHandlerContract.handleFees.value(tradeData.networkFeeWei)(eligibleReserves, rebatePercentages),
+            "Transfer network fee to FeeHandler failed"
+        );
+        emit HandleFees(address(feeHandlerContract), tradeData.networkFeeWei);
+
+
+    }
+
+    function updateEligibilityAndRebates(
+        address[] memory eligibleReserves,
+        uint[] memory rebatePercentages,
+        TradeData memory tradeData
+    ) internal pure returns(uint) {
```

Comment:
please start new function in new line

{ in new line

Reply:
done